pkg/user/store/userstore.go

```diff
@@ -0,0 +1,47 @@
+package store
+
+import (
+	"github.com/go-pg/pg/v9"
+
+	model "github.com/chrsep/vor/pkg/postgres"
+)
+
+type (
+		UserStore interface {
+			GetUser(userID string) (*model.User, error)
+			GetSchools(userID string) ([]model.School, error)
+	}
```

Comment:
What are the advantages of having store inside of its own package together with its interfaces?

I think its better for interfaces to live as close as possible to where it is being used. That way we'll have an easier time finding more context of how it's being used. Also, the store interface was planned to be used as a way to completely separate packages from the DB/storage implementation (I've finally had a package that does this correctly, the new `class` package). 

For example, the `class` package that I just added does not have any dependency or knowledge of how we store the data. Wouldn't having the interface here do the opposite? Forcing the `user` package to have an indirect dependency on `Postgres` package?

Reply:
but if we use like that, will that increase complexity? for ex: currently your store method implementation is in postgres package but the interface is in user package. if we want to change the implementation of the method and param, we need to search of where the file of interface and method implementation and change those

what do you mean by indirect dependency on postgres package? and `class` package receive store param as argument in one of their method right, does this mean `class` package also have dependency on postgres package?

Reply:
I don't think separating implementation from the interface would increase complexity. You could just use the "go to implementation" shortcut on your editor to browse/jump to the interface's implementations. Besides, interfaces were meant to be implemented by other packages anyway right? Acting like some sort of universal connector.

Reply:
> what do you mean by indirect dependency on postgres package?
The `class` package can be used without the `postgres` package. We could delete the `postgres` package,  And implement another store in `mysql` package, or `s3` package. We could plug those into `class` package, and everything else will work just fine.

The `user` package on the other hand, can't function without the `postgres` package. If we delete that, the code won't compile.

Reply:
ah yes, gw baru cek class package. maksud gw kaya gini storenya di dalam 1 package. bedanya apa sama user?

Reply:
so you suggest `user` package to be refactor like `class` package?